File: 2022/day3/ocaml/lib/part1.ml
let calculate_line line =
    let comp_size = (String.length line) / 2 in
    let first = String.sub line 0 comp_size
    |> Common.str_to_set_of_chars in
    let second = String.sub line comp_size comp_size
    |> Common.str_to_set_of_chars in

    Common.CS.inter first second
    |> Common.CS.choose
    |> Common.get_priority

let solve ic =
    let rec s acc =
        match input_line ic with
        | line -> s (acc + (calculate_line line))
        | exception End_of_file -> acc in
    s 0

File: 2022/day3/cpp/part1.cpp
#include <fstream>
#include <string>
#include <vector>
#include "part1.h"
#include "common.h"

int handleLine(std::string *line) {
    int compSize = line->size() / 2;
    std::vector<char> a(line->c_str(), line->c_str() + compSize);
    std::vector<char> b(line->c_str() + compSize, line->c_str() + compSize + compSize);
    char ch = getCommonCharacter({a, b});
    return getCharPriority(ch);
}

int part1Solve(std::ifstream *file) {
    int acc = 0;
    std::string line;
    while (std::getline(*file, line)) {
        acc += handleLine(&line);
    }
    return acc;
}
